src/traces/bar/calc.js

```diff
@@ -17,7 +17,9 @@ var calcSelection = require('../scatter/calc_selection');
 module.exports = function calc(gd, trace) {
```

Comment:
Thanks for reusing almost :sparkle: **all** :sparkle: of `Bar`.

Now, I would prefer if we use an inheritance pattern more similar to how `barpolar` calls `Bar` methods:

https://github.com/plotly/plotly.js/blob/02e51d8f5ba5601f53ee3dc839ba05fb12c7f8a2/src/traces/barpolar/calc.js#L76-L104

instead of having `type === 'waterfall'` clauses inside `traces/bar/`. In other words, `traces/bar/` `shouldn't have to know anything about waterfall traces.